### PR TITLE
Optimize count with an overload which calls walkLength

### PR DESCRIPTION
**Pull request by RazvanN7**
Following the discussion in #5545 , I created this separate PR for the count optimization. The implementation uses walkLength to compute the count. If walkLength will be deprecated, then I will replace the call with the code, otherwise we can close the other PR and keep just this oprimization

### CODE REVIEW

**Comment by UplinkCoder** [comment]
Why is that needed.

The ddoc is bigger then the function ?
which is a thunk.
a templated thunk no less.
introducing symbol bloat.
You could just have made an alias you know ...

**Comment by wilzbach** [comment]
> You could just have made an alias you know ...

Would break code, due to the different default behavior. `count` allows a user-defined `pred`, `walkLength` allows a maximal `count`.
Though imho on the long run on of these functions should be deprecated (or at least removed from the docs).

> The ddoc is bigger then the function ?

Because documentation is __a lot__ more important than code. This is a common pattern in Phobos ;-)

**Comment by RazvanN7** [comment]
@wilzbach I created an overload anyway so making an alias won't break any code, but it seems cleaner to me to have an overload, rather then an alias.

**Comment by wilzbach** [comment]
(keeping this open for one or two days to give people enough time to veto this approach as a general good practice)

File: std/algorithm/searching.d [hunk 1]
```diff
@@ -604,6 +604,11 @@ $(D 2).
 The third version counts the elements for which $(D pred(x)) is $(D
 true). Performs $(BIGOH haystack.length) evaluations of $(D pred).
 
+The fourth version counts the number of elements in a range. It is
+an optimization for the third version: if the given range has the
+length property the count is returned right away, otherwise
```
Comment (andralex) [inline std/algorithm/searching.d]: `length` in code font (e.g. backticks)

**Comment by andralex** [comment]
Aliases are generally better, but here we have overloading so that wouldn't work well.

This improves the situation whereby someone calls count with the default predicate - avoid the goofy walking the whole range. That's fine, as is to allow two distinct library functions sometimes do the same thing under certain circumstances.

**Comment by RazvanN7** [comment]
@andralex done

**Pull request merged**